### ISSUE TITLE
"For you" content should be by user preference customizable

### ISSUE THREAD

**Issue by d-akara** [twitter/the-algorithm#1363]
Essentially replace all the weighting with a set of preferences that the user can opt-in or opt-out.
Let the user customize their own experience instead of a set of algorithms that are applied to everyone.

For example of some suggested ideas for options:

- select original tweets only vs retweets
- tweets from my followers, follows tweets. 
- popular tweets my followers like, comment.
- tweets related to my recent likes [ specify within last n days ]
- tweets related to my recent tweets [ specify within last n days ]
- tweets trending in categories of my interest [ I can specify categories or tags etc ]
- exclude tweets [ user can specify spam criteria or categories or content they don't want to see ]
- rank tweets by [ user can specify, likes, retweets, comments, etc ]

**Comment by alizardguy** [comment]
More transparency and customization in the "For you" content would be very pro consumer and would line up with what Elon claimed his goals for Twitter were.

**Comment by DinhHuynhTienPhu** [comment]
Yes. This is what a social web need. User should have more control on what they want to see.

**Comment by Christophvh** [comment]
Agree, but a general 'sensible' default would be needed, because a lot of people never use options/filters/... they just want it to work instantly, and that is why TikTok is such a huge success.

**Comment by d-akara** [comment]
Yes, defaults make sense and make it opt-in. Its use would grow organically as people tweet and comment how to build better views for content etc.
